src/stm32/n32g45x_adc.c

```diff
@@ -0,0 +1,179 @@
+// ADC functions on N32G45x
+//
+// Copyright (C) 2019-2020  Kevin O'Connor <kevin@koconnor.net>
+// Copyright (C) 2022-2023  Alexey Golyshin <stas2z@gmail.com>
+//
+// Based on Kevin's stm32f1x code, but modified for N32G45x specific ADC
```

Comment:
It's preferable not to put file history in the code comments - the git commit history can be used for that.  It's not necessary to credit me here.

Also, if you do have code comments, it's preferable to keep them separate from the initial license block, as we want to keep the license info succinct.